src/extension.ts

```diff
@@ -157,9 +176,9 @@ export function noWorkspaceFolder(uri: Uri): WorkspaceFolder {
 }
 
 export function isCrystalDocument(doc: TextDocument): boolean {
-    return doc.languageId === 'crystal'
+    return doc.languageId === 'crystal' && (doc.uri.scheme === 'file' || doc.uri.scheme === 'untitled')
```

Comment:
What's the reason of this change? Also, that's not what the name of the function suggest.

Reply:
We only want to support these two schemas, as otherwise it creates a mess locally (looking at git changes, will duplicate all the lints, which is annoying).

Reply:
So what's the reason for the _mess_?

Reply:
Feel free to make any commits to this PR to get it exactly how you want it.

Reply:
Thanks, I'm trying to understand the reasoning behind these changes.

Reply:
Like I said, it's to only support the file and untitled schemas. I don't see a problem with limiting the schema checks within this method, as otherwise would require adding an additional method for checking the schema to everywhere this method is already called. This method was also originally called something to the effect of "isValidCrystalDocument" until y'all had me change it.